### PR TITLE
Inferred return types

### PR DESCRIPTION
**Pull request by dirk**
This introduced inferring of return types for functions. It also adds:

1. `else` blocks following an `if`
2. Proper `Void` type in the type-system

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ language: node_js
 node_js:
   - "0.12"
 before_script:
-  - "npm run grammar2"
+  - "npm run grammar"
```
Comment (ryngonzalez) [inline .travis.yml]: yay!

Comment (dirk) [inline .travis.yml]: :vacuum:

File: lib/grammar.pegjs [hunk 2]
```diff
@@ -72,7 +72,8 @@ ctrl = ifctrl
      / forctrl
      / returnctrl
 
-ifctrl     = "if" _ c:innerstmt _ b:block { return p.parseIf(c, b) }
+ifctrl     = "if" _ c:innerstmt _ b:block e:(_ elsecont)? { return p.parseIf(c, b, e ? e[1] : null) }
+elsecont   = "else" _ b:block { return b }
```
Comment (ryngonzalez) [inline lib/grammar.pegjs]: Is `else if` defined yet?

Comment (dirk) [inline lib/grammar.pegjs]: Not yet. Just made #21 for it.

**Pull request merged**